--- ./vacation.c.mkbackup	Thu Mar 23 13:04:51 2000
+++ ./vacation.c	Tue Oct  3 23:05:18 2000
@@ -573,15 +573,35 @@
 {
   char *editor;
   char ebuf[PATH_MAX];
+  char c;
   FILE *message;
+  FILE *oldmessage;
   FILE *forward;
+  FILE *oldforward;
 
-  if (((forward = fopen (FWD, "w")) == NULL)) {
-    fprintf (stderr, "vacation: cannot open %s\n", FWD);
-    exit (1);
+  if (((forward = fopen (FWD, "r")) == NULL)) {
+    if (((forward = fopen (FWD, "w")) == NULL)) {
+      fprintf (stderr, "vacation: cannot open %s for reading\n", FWD);
+      exit (1);
+    }
+  } else {
+    if (((oldforward = fopen (OLDFWD, "w")) == NULL)) {
+      fprintf (stderr, "vacation: cannot open %s for writing\n", OLDFWD);
+      exit (1);
+    }
+    while (((c = fgetc (forward)) != EOF)) {
+      fprintf(oldforward, "%c", c);
+      chmod (OLDFWD, 00600);
+    }
+    fclose (oldforward);
+    if (((forward = freopen (FWD, "w", forward)) == NULL)) {
+      fprintf (stderr, "vacation: cannot open %s for writing\n", FWD);
+      exit (1);
+    }
   }
   fprintf (forward, "\\%s, \"|%s %s\"\n", myname, path, myname);
   fclose (forward);
+  chmod (FWD, 00600);
 
   if (((editor = getenv ("VISUAL")) == NULL))
     if (((editor = getenv ("EDITOR")) == NULL))
@@ -591,14 +611,26 @@
 	printd (logline);
 #endif
         
-  if (((message = fopen (VMSG, "w")) == NULL)) {
-    fprintf (stderr, "vacation: cannot open %s\n", VMSG);
-    exit (1);
-  }
+  if (((message = fopen (VMSG, "r")) == NULL)) {
+    if (((message = fopen (VMSG, "w")) == NULL)) {
+      fprintf (stderr, "vacation: cannot open %s\n", VMSG);
+      exit (1);
+    }
+  message = fopen (VMSG, "w");
   fprintf (message, "Subject: away from my mail\n\n");
   fprintf (message, "I will not be reading my mail for a while.\n");
   fprintf (message, "Your mail concerning \"$SUBJECT\"\n");
   fprintf (message, "will be read when I'm back.\n");
+  } else {
+    if (((oldmessage = fopen (OLDVMSG, "w")) == NULL)) {
+      fprintf (stderr, "vacation: cannot open %s\n", OLDVMSG);
+      exit (1);
+    }
+    while (((c = fgetc (message)) != EOF)) {
+      fprintf(oldmessage, "%c", c);
+    }
+    fclose (oldmessage);
+  }
   fclose (message);
   sprintf (ebuf, "%s %s", editor, VMSG);
 #ifdef DEBUG
--- ./vacation.h.mkbackup	Thu Mar 23 13:04:51 2000
+++ ./vacation.h	Tue Oct  3 22:39:01 2000
@@ -12,7 +12,9 @@
 #define	MAXLINE	1024			/* max line from mail header */
 #define	VDB	".vacation.db"		/* dbm's database */
 #define	VMSG	".vacation.msg"		/* vacation message */
+#define	OLDVMSG	".vacation.old"		/* previous vacation message */
 #define FWD     ".forward"              /* forward file */
+#define OLDFWD  ".forward.old"          /* previous forward file */
 
 typedef struct alias {
 	struct alias *next;
--- ./vacation.man.mkbackup	Thu Mar 23 13:04:53 2000
+++ ./vacation.man	Tue Oct  3 22:39:01 2000
@@ -57,9 +57,13 @@
 .\"by default).
 The
 .B vacation
-program run interactively will create a 
+program run interactively will create and/or edit a 
 .BR .vacation.msg
-file for you (which you may edit).  
+file in your home directory. The old
+.BR .vacation.msg
+will be backed up to
+.BR .vacation.old
+file.
 Type 
 .B
 vacation
@@ -199,7 +203,7 @@
 issue short usage line.
 .SH USAGE
 .LP
-To start
+The
 .BR vacation ,
 create a
 .B \&.forward
@@ -209,7 +213,12 @@
 .LP
 where
 .I username
-is your login name.
+is your login name. The original
+.B \&.forward
+will be backed up to
+.B \&.forward.old
+file.
+
 .LP
 Then type in the command:
 .IP
@@ -262,5 +271,3 @@
 Berkeley, California, and Copyright (c) 1993 by Harald Milz
 (hm@seneca.ix.de). Tiny patches 1998 by Mark Seuffert (moak@pirate.de).
 Now maintained by Sean Rima (thecivvie@softhome.net)
-
-
